xonsh/history/base.py

```diff
@@ -164,3 +165,20 @@ def clear(self):
         memory.
         """
         pass
+
+    def is_ignored(self, cmd):
+        """Determines if a history item should be added to the event history.
+        Call this in your append method.
+
+        Parameters
+        ----------
+        cmd: The prospective item to append.
+
+        Returns
+        -------
+        True if the item should be appended, False if not.
+        """
```

Comment:
please check the [numpy-docstring](https://numpydoc.readthedocs.io/en/latest/format.html#parameters) format for parameters. The description should appear in the next line with indent